diff --git a/config.def.h b/config.def.h
index 546edda..d2d08f2 100644
--- a/config.def.h
+++ b/config.def.h
@@ -82,6 +82,9 @@ char *termname = "st-256color";
  */
 unsigned int tabspaces = 8;
 
+float alpha = 0.8; /* Better Alpha */
+float alphaNoFocus = 0.6; /* Better Alpha */
+
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
 	/* 8 normal colors */
diff --git a/config.mk b/config.mk
index 0cbb002..be85d54 100644
--- a/config.mk
+++ b/config.mk
@@ -16,7 +16,8 @@ PKG_CONFIG = pkg-config
 INCS = -I$(X11INC) \
        `$(PKG_CONFIG) --cflags fontconfig` \
        `$(PKG_CONFIG) --cflags freetype2`
-LIBS = -L$(X11LIB) -lm -lrt -lX11 -lutil -lXft \
+# Better Alpha #
+LIBS = -L$(X11LIB) -lm -lrt -lX11 -lutil -lXft -lXrender \
        `$(PKG_CONFIG) --libs fontconfig` \
        `$(PKG_CONFIG) --libs freetype2`
 
diff --git a/st.1 b/st.1
index e8d6059..43466a9 100644
--- a/st.1
+++ b/st.1
@@ -4,6 +4,10 @@ st \- simple terminal
 .SH SYNOPSIS
 .B st
 .RB [ \-aiv ]
+.RB [ \-q\" Better Alpha
+.IR alpha ]
+.RB [ \-Q\" Better Alpha
+.IR alpha ]
 .RB [ \-c
 .IR class ]
 .RB [ \-f
@@ -28,6 +32,10 @@ st \- simple terminal
 .PP
 .B st
 .RB [ \-aiv ]
+.RB [ \-q\" Better Alpha
+.IR alpha ]
+.RB [ \-Q\" Better Alpha
+.IR alpha ]
 .RB [ \-c
 .IR class ]
 .RB [ \-f
@@ -55,6 +63,12 @@ is a simple terminal emulator.
 .B \-a
 disable alternate screens in terminal
 .TP
+.BI \-q " alpha"\" Better Alpha
+defines the alpha value when st has focus
+.TP
+.BI \-Q " alpha"\" Better Alpha
+defines the alpha value when st doesn't have focus
+.TP
 .BI \-c " class"
 defines the window class (default $TERM).
 .TP
diff --git a/st.h b/st.h
index a1928ca..acbc30e 100644
--- a/st.h
+++ b/st.h
@@ -121,3 +121,5 @@ extern char *termname;
 extern unsigned int tabspaces;
 extern unsigned int defaultfg;
 extern unsigned int defaultbg;
+extern float alpha; /* Better Alpha */
+extern float alphaNoFocus; /* Better Alpha */
diff --git a/x.c b/x.c
index 4cf6b21..6c2aecf 100644
--- a/x.c
+++ b/x.c
@@ -105,6 +105,7 @@ typedef struct {
 	XSetWindowAttributes attrs;
 	int scr;
 	int isfixed; /* is fixed geometry? */
+	int depth; /* bit depth */ /* Better Alpha */
 	int l, t; /* left and top offset */
 	int gm; /* geometry mask */
 } XWindow;
@@ -243,6 +244,9 @@ static char *usedfont = NULL;
 static double usedfontsize = 0;
 static double defaultfontsize = 0;
 
+static int focused            = 1;    /* Better Alpha */
+static char *opt_alpha        = NULL; /* Better Alpha */
+static char *opt_alphaNoFocus = NULL; /* Better Alpha */
 static char *opt_class = NULL;
 static char **opt_cmd  = NULL;
 static char *opt_embed = NULL;
@@ -734,7 +738,7 @@ xresize(int col, int row)
 
 	XFreePixmap(xw.dpy, xw.buf);
 	xw.buf = XCreatePixmap(xw.dpy, xw.win, win.w, win.h,
-			DefaultDepth(xw.dpy, xw.scr));
+			xw.depth); /* Better Alpha */
 	XftDrawChange(xw.draw, xw.buf);
 	xclear(0, 0, win.w, win.h);
 
@@ -772,6 +776,23 @@ xloadcolor(int i, const char *name, Color *ncolor)
 	return XftColorAllocName(xw.dpy, xw.vis, xw.cmap, name, ncolor);
 }
 
+/* Better Alpha */
+void
+xloadalpha(void)
+{
+	/* set alpha value of bg color */
+	if (opt_alpha)
+		alpha = strtof(opt_alpha, NULL);
+	if (opt_alphaNoFocus)
+		alphaNoFocus = strtof(opt_alphaNoFocus, NULL);
+
+	const float usedAlpha = focused ? alpha : alphaNoFocus;
+
+	dc.col[defaultbg].color.alpha = (unsigned short)(0xffff * usedAlpha);
+	dc.col[defaultbg].pixel &= 0x00FFFFFF;
+	dc.col[defaultbg].pixel |= (unsigned char)(0xff * usedAlpha) << 24;
+}
+
 void
 xloadcols(void)
 {
@@ -794,6 +815,7 @@ xloadcols(void)
 			else
 				die("could not allocate color %d\n", i);
 		}
+	xloadalpha(); /* Better Alpha */
 	loaded = 1;
 }
 
@@ -1103,11 +1125,25 @@ xinit(int cols, int rows)
 	Window parent;
 	pid_t thispid = getpid();
 	XColor xmousefg, xmousebg;
+	XWindowAttributes attr; /* Better Alpha */
+	XVisualInfo vis; /* Better Alpha */
 
 	if (!(xw.dpy = XOpenDisplay(NULL)))
 		die("can't open display\n");
 	xw.scr = XDefaultScreen(xw.dpy);
-	xw.vis = XDefaultVisual(xw.dpy, xw.scr);
+
+	/* ++Better Alpha */
+	if (!(opt_embed && (parent = strtol(opt_embed, NULL, 0)))) {
+		parent = XRootWindow(xw.dpy, xw.scr);
+		xw.depth = 32;
+	} else {
+		XGetWindowAttributes(xw.dpy, parent, &attr);
+		xw.depth = attr.depth;
+	}
+
+	XMatchVisualInfo(xw.dpy, xw.scr, xw.depth, TrueColor, &vis);
+	xw.vis = vis.visual;
+	/* --Better Alpha */
 
 	/* font */
 	if (!FcInit())
@@ -1117,7 +1153,7 @@ xinit(int cols, int rows)
 	xloadfonts(usedfont, 0);
 
 	/* colors */
-	xw.cmap = XDefaultColormap(xw.dpy, xw.scr);
+	xw.cmap = XCreateColormap(xw.dpy, parent, xw.vis, None); /* Better Alpha */
 	xloadcols();
 
 	/* adjust fixed window geometry */
@@ -1138,18 +1174,16 @@ xinit(int cols, int rows)
 	xw.attrs.colormap = xw.cmap;
 
 	if (!(opt_embed && (parent = strtol(opt_embed, NULL, 0))))
-		parent = XRootWindow(xw.dpy, xw.scr);
+		parent = XRootWindow(xw.dpy, xw.scr); /* This if is deleted in the default alphaFocusHighlight */
 	xw.win = XCreateWindow(xw.dpy, parent, xw.l, xw.t,
-			win.w, win.h, 0, XDefaultDepth(xw.dpy, xw.scr), InputOutput,
+			win.w, win.h, 0, xw.depth, InputOutput, /* Better Alpha */
 			xw.vis, CWBackPixel | CWBorderPixel | CWBitGravity
 			| CWEventMask | CWColormap, &xw.attrs);
 
 	memset(&gcvalues, 0, sizeof(gcvalues));
 	gcvalues.graphics_exposures = False;
-	dc.gc = XCreateGC(xw.dpy, parent, GCGraphicsExposures,
-			&gcvalues);
-	xw.buf = XCreatePixmap(xw.dpy, xw.win, win.w, win.h,
-			DefaultDepth(xw.dpy, xw.scr));
+	xw.buf = XCreatePixmap(xw.dpy, xw.win, win.w, win.h, xw.depth); /* Better Alpha */
+	dc.gc = XCreateGC(xw.dpy, xw.buf, GCGraphicsExposures, &gcvalues); /* Better Alpha */
 	XSetForeground(xw.dpy, dc.gc, dc.col[defaultbg].pixel);
 	XFillRectangle(xw.dpy, xw.buf, dc.gc, 0, 0, win.w, win.h);
 
@@ -1730,12 +1764,22 @@ focus(XEvent *ev)
 		xseturgency(0);
 		if (IS_SET(MODE_FOCUS))
 			ttywrite("\033[I", 3, 0);
+		if (!focused) { /* Better Alpha */
+			focused = 1;
+			xloadalpha();
+			redraw();
+		}
 	} else {
 		if (xw.ime.xic)
 			XUnsetICFocus(xw.ime.xic);
 		win.mode &= ~MODE_FOCUSED;
 		if (IS_SET(MODE_FOCUS))
 			ttywrite("\033[O", 3, 0);
+		if (focused) { /* Better Alpha */
+			focused = 0;
+			xloadalpha();
+			redraw();
+		}
 	}
 }
 
@@ -1973,12 +2017,12 @@ run(void)
 void
 usage(void)
 {
-	die("usage: %s [-aiv] [-c class] [-f font] [-g geometry]"
-	    " [-n name] [-o file]\n"
+	die("usage: %s [-aiv] [-q alpha] [-Q alpha] [-c class] [-f font]"
+	    " [-g geometry] [-n name] [-o file]\n"
 	    "          [-T title] [-t title] [-w windowid]"
 	    " [[-e] command [args ...]]\n"
-	    "       %s [-aiv] [-c class] [-f font] [-g geometry]"
-	    " [-n name] [-o file]\n"
+	    "       %s [-aiv] [-q alpha] [-Q alpha] [-c class] [-f font]"
+	    " [-g geometry] [-n name] [-o file]\n"
 	    "          [-T title] [-t title] [-w windowid] -l line"
 	    " [stty_args ...]\n", argv0, argv0);
 }
@@ -1994,6 +2038,12 @@ main(int argc, char *argv[])
 	case 'a':
 		allowaltscreen = 0;
 		break;
+	case 'q': /* Better Alpha */
+		opt_alpha = EARGF(usage());
+		break;
+	case 'Q': /* Better Alpha */
+		opt_alphaNoFocus = EARGF(usage());
+		break;
 	case 'c':
 		opt_class = EARGF(usage());
 		break;
